test/appium/tests/atomic/chats/test_public.py

```diff
@@ -437,23 +437,72 @@ def test_different_status_in_timeline(self):
+        home_1.get_back_to_home_view()
+
+        home_1.just_fyi("Creating group chat and sending message from device 1")
+        group_chat_1 = home_1.create_group_chat([username_2], group_chat_name)
+        group_chat_1.send_message("HI")
+        home_1.get_back_to_home_view()
+
+        home_1.just_fyi("Creating public chat and sending message from device 1")
+        pub_chat_1, pub_chat_2 = home_1.join_public_chat(pub_chat_name), home_2.join_public_chat(pub_chat_name)
+        pub_chat_1.send_message("HI")
         device_1.toggle_airplane_mode()
-        chat_2.send_message(message)
+
+        home_2.just_fyi("Joining public chat by device 2 and sending message")
+        pub_chat_2.send_message(message_1)
+        home_2.get_back_to_home_view()
+
+        home_2.just_fyi("Joining 1-1 chat by device 2 and sending message")
+        one_to_one_chat_element = home_2.get_chat(username_1, wait_time=10)
+        one_to_one_chat_2 = one_to_one_chat_element.click()
+        one_to_one_chat_2.send_message(message_2)
+        home_2.get_back_to_home_view()
+
+        home_2.just_fyi("Joining Group chat by device 2 and sending message")
+        home_2.notifications_button.click()
+        group_chat_2 = home_2.get_chat_from_activity_center_view(group_chat_name).click()
+        group_chat_2.join_chat_button.click()
+        group_chat_2.send_message(message_2)
+
+        # Waiting for 3 minutes and then going back online
         sleep(180)
         device_1.toggle_airplane_mode()
-        chat_1.element_by_translation_id("fetch-messages").wait_and_click(60)
-        if not chat_1.chat_element_by_text(message).is_element_displayed():
-            device_1.driver.fail("Test message has not been fetched")
+
+        home_1.just_fyi("Checking gap in public chat and fetching messages")
+        if pub_chat_1.chat_element_by_text(message_1).is_element_displayed(10):
+            self.errors.append("Test message has been fetched automatically")
+        pub_chat_1.element_by_translation_id("fetch-messages").wait_and_click(60)
+        if not pub_chat_1.chat_element_by_text(message_1).is_element_displayed(10):
+            self.errors.append("Test message has not been fetched")
+        home_1.get_back_to_home_view()
+
+        home_1.just_fyi("Checking that there is no gap in 1-1 chat and messages fetched automatically")
+        home_1.get_chat(username_2).click()
+        if not one_to_one_chat_1.chat_element_by_text(message_2).is_element_displayed(10):
+            self.errors.append("Message in 1-1 chat has not been fetched automatically")
+        home_1.get_back_to_home_view()
+
+        home_1.just_fyi("Checking that there is no gap in group chat and messages fetched automatically")
+        home_2.get_chat(group_chat_name).click()
+        if not group_chat_1.chat_element_by_text(message_2).is_element_displayed(10):
+            self.errors.append("Message in group chat has not been fetched automatically")
```

Comment:
self.errors.verify_no_errors() at the end, otherwise test will just pass even with errors